File: VB-Examples/04_RowsColumns/HideRowsAndColumns.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls

Namespace HideRowsAndColumns
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			Dim workbook As New Workbook()
			workbook.LoadFromFile("..\..\..\..\..\..\Data\HideRowsAndColumns.xls")

			Dim worksheet As Worksheet = workbook.Worksheets(0)
			' Hiding the column of the worksheet
			worksheet.HideColumn(2)
			'Hiding the row of the worksheet
			worksheet.HideRow(4)

			workbook.SaveToFile("HideRowsAndColumns.xlsx", ExcelVersion.Version2010)

			ExcelDocViewer("HideRowsAndColumns.xlsx")
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class
End Namespace

File: CS-Examples/04_RowsColumns/HideRowsAndColumns.cs
using System;
using System.Data.OleDb;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;
using System.Data;

using Spire.Xls;

namespace HideRowsAndColumns
{
	public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
		private void btnRun_Click(object sender, System.EventArgs e)
		{
            // Create a new workbook object
            Workbook workbook = new Workbook();

            // Load an existing Excel file named "HideRowsAndColumns.xls"
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\HideRowsAndColumns.xls");

            // Get the first worksheet from the workbook
            Worksheet worksheet = workbook.Worksheets[0];

            // Hide the second column of the worksheet
            worksheet.HideColumn(2);

            // Hide the fourth row of the worksheet
            worksheet.HideRow(4);

            // Save the modified workbook to a new file named "HideRowsAndColumns.xlsx" in Excel 2010 format
            workbook.SaveToFile("HideRowsAndColumns.xlsx", ExcelVersion.Version2010);

            // View the file
            ExcelDocViewer("HideRowsAndColumns.xlsx");
		}
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
	}
}
